CharlieBackend.Data/Repositories/Impl/StudentGroupRepository.cs

```diff
@@ -114,5 +116,16 @@ public async Task<List<MentorStudyGroupsDto>> GetMentorStudyGroups(long id)
             return await _applicationContext.StudentsOfStudentGroups.Where(s => s.StudentGroupId == id)
                 .Select(s => s.StudentId).ToListAsync();
         }
+
+        public async Task<IList<StudentGroup>> GetStudentGroupsByDateAsync(DateTime startDate,DateTime finishDate)
+        {
+            return await _applicationContext.StudentGroups
+                .AsNoTracking()
+                .Where(x => x.StartDate < finishDate && x.StartDate >= startDate || 
+                       x.FinishDate > startDate && x.FinishDate <= finishDate)
```

Comment:
If we ever need to check *if the given date is in range*, it could be useful to implement this logic in a separate extension method.